### PR TITLE
Tickets/dm 33508

### PR DESCRIPTION
**Pull request by tribeiro**

### CODE REVIEW

File: tests/test_make_idl_file.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        if os.path.exists(idl_file_path):
+            os.remove(idl_file_path)
+
+        make_idl_file(
+            name=component,
+            keep_all=False,
+        )
+
+        assert os.path.exists(idl_file_path)
```
Comment (r-owen) [inline tests/test_make_idl_file.py:44]: It makes me nervous to have this test permanently modify data in ts_idl. What if the developer is intentionally using an older version?

Pleases consider move the old file aside, if it exists, then moving it back in a `finally` clause.

The trick, I suppose, is deciding how to move the file. Perhaps using `tempfile` library would be a reasonable solution for that -- e.g. by creating a temporary directory, moving the old file there, then moving it back. The directory would be automatically cleaned up as long as you used tempfile.TemporaryDirectory as a context manager.

File: python/lsst/ts/sal/make_idl_file.py [hunk 2]
```diff
@@ -55,17 +55,6 @@ def __init__(self, name, keep_all):
...
-        interfaces_dir = os.path.join(self.xml_dir, "sal_interfaces")
-        std_xml_paths = [
-            os.path.join(interfaces_dir, f"{n}.xml")
-            for n in ("SALGenerics", "SALSubsystems")
-        ]
-        sal_xml_paths = glob.glob(os.path.join(interfaces_dir, self.name, "*.xml"))
-
-        for xmlpath in std_xml_paths + sal_xml_paths:
-            shutil.copy(xmlpath, self.sal_work_dir)
-
```
Comment (r-owen) [inline python/lsst/ts/sal/make_idl_file.py:68]: I'm happy to see this isn't needed.

Should delete_files be updated? There's no point if ts_sal's salgenerator is copying the xml files. But if the xml files are being read in place, are there still xml files to delete at line 71?

**Review by r-owen — APPROVED** [review]
Please consider not overwriting an existing idl file. Otherwise this is a very welcome addition.

File: python/lsst/ts/sal/make_idl_file.py (hunk 2 above)
Comment (dmills-lsst) [reply python/lsst/ts/sal/make_idl_file.py:68]: salgenerator will generate the <CSC>_Generics.xml in the $SAL_WORK_DIR and use the others from ts_xml directly

**Pull request merged**